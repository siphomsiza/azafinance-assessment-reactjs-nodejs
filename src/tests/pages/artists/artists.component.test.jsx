import React from 'react';
import { shallow } from 'enzyme';
import MyComponent from '../../../pages/artists/artists.component';
describe("Page Artists", () => {
  it("should render Artists component", () => {
    const wrapper = shallow(<MyComponent />);
    //console.log(wrapper.instance())   
    expect(wrapper.find('div')).toMatchSnapshot();
  });

});
