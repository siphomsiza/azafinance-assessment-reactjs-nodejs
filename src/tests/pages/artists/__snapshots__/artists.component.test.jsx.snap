// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Component should render my component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="shop-page"
  >
    <React.Fragment>
      <SearchBar
        search={[Function]}
      />
      <CollectionPreview
        data={Array []}
      />
    </React.Fragment>
  </div>,
  "nodes": Array [
    <div
      className="shop-page"
    >
      <React.Fragment>
        <SearchBar
          search={[Function]}
        />
        <CollectionPreview
          data={Array []}
        />
      </React.Fragment>
    </div>,
  ],
  "options": Object {},
  "renderer": null,
  "root": ShallowWrapper {
    "complexSelector": ComplexSelector {
      "buildPredicate": [Function],
      "childrenOfNode": [Function],
      "findWhereUnwrapped": [Function],
    },
    "length": 1,
    "node": <div
      className="shop-page"
    >
      <React.Fragment>
        <SearchBar
          search={[Function]}
        />
        <CollectionPreview
          data={Array []}
        />
      </React.Fragment>
    </div>,
    "nodes": Array [
      <div
        className="shop-page"
      >
        <React.Fragment>
          <SearchBar
            search={[Function]}
          />
          <CollectionPreview
            data={Array []}
          />
        </React.Fragment>
      </div>,
    ],
    "options": Object {},
    "renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useDeferredValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useResponder": [Function],
        "useState": [Function],
        "useTransition": [Function],
      },
      "_element": <Home />,
      "_firstWorkInProgressHook": Object {
        "memoizedState": Object {
          "artists": Array [],
          "errorMessage": null,
          "loading": false,
        },
        "next": null,
        "queue": Object {
          "dispatch": [Function],
          "first": null,
        },
      },
      "_forcedUpdate": false,
      "_instance": null,
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div
        className="shop-page"
      >
        <React.Fragment>
          <SearchBar
            search={[Function]}
          />
          <CollectionPreview
            data={Array []}
          />
        </React.Fragment>
      </div>,
      "_rendering": false,
      "_updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
      "_workInProgressHook": null,
      "getRenderOutput": [Function],
      "render": [Function],
    },
    "root": [Circular],
    "unrendered": <Home />,
  },
  "unrendered": null,
}
`;

exports[`Page Artists should render Artists component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="shop-page"
  >
    <React.Fragment>
      <SearchBar
        search={[Function]}
      />
      <CollectionPreview
        data={Array []}
      />
    </React.Fragment>
  </div>,
  "nodes": Array [
    <div
      className="shop-page"
    >
      <React.Fragment>
        <SearchBar
          search={[Function]}
        />
        <CollectionPreview
          data={Array []}
        />
      </React.Fragment>
    </div>,
  ],
  "options": Object {},
  "renderer": null,
  "root": ShallowWrapper {
    "complexSelector": ComplexSelector {
      "buildPredicate": [Function],
      "childrenOfNode": [Function],
      "findWhereUnwrapped": [Function],
    },
    "length": 1,
    "node": <div
      className="shop-page"
    >
      <React.Fragment>
        <SearchBar
          search={[Function]}
        />
        <CollectionPreview
          data={Array []}
        />
      </React.Fragment>
    </div>,
    "nodes": Array [
      <div
        className="shop-page"
      >
        <React.Fragment>
          <SearchBar
            search={[Function]}
          />
          <CollectionPreview
            data={Array []}
          />
        </React.Fragment>
      </div>,
    ],
    "options": Object {},
    "renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useDeferredValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useResponder": [Function],
        "useState": [Function],
        "useTransition": [Function],
      },
      "_element": <Home />,
      "_firstWorkInProgressHook": Object {
        "memoizedState": Object {
          "artists": Array [],
          "errorMessage": null,
          "loading": false,
        },
        "next": null,
        "queue": Object {
          "dispatch": [Function],
          "first": null,
        },
      },
      "_forcedUpdate": false,
      "_instance": null,
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div
        className="shop-page"
      >
        <React.Fragment>
          <SearchBar
            search={[Function]}
          />
          <CollectionPreview
            data={Array []}
          />
        </React.Fragment>
      </div>,
      "_rendering": false,
      "_updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
      "_workInProgressHook": null,
      "getRenderOutput": [Function],
      "render": [Function],
    },
    "root": [Circular],
    "unrendered": <Home />,
  },
  "unrendered": null,
}
`;
